core/core-documentation.el

```diff
@@ -85,6 +85,52 @@
         (format "%s\n%s%s" beginning-of-content toc-string rest-of-content)
       content)))
 
+(defun spacemacs//spacemacs-doc-publishing-function (plist filename pub-dir)
+  (progn
+    (find-file filename)
+    (unless (re-search-forward "\+HTML_HEAD_EXTRA\:.*\/css\/readtheorg.css" nil t nil)
+      (let ((head-css-extra-readtheorg (concat
+                                        "#+HTML_HEAD_EXTRA:"
+                                        "<link rel=\"stylesheet\" "
+                                        "type=\"text/css\" "
+                                        "href=\"../css/readtheorg.css\" />\n")))
+        (progn
+          (setq buffer-read-only nil)
```

Comment:
As an alternative we can set `inhibit-read-only` before starting the publishing process and turn it off afterwards.